### PR TITLE
More logging improvements

### PR DESCRIPTION
**Pull request by hawkowl**
This implements:
- Loglevels being applied to log file output (Py2 only, can't write to log output on Py3)
- Worker containers communicating logs by JSON through stdout, to give richer and multi-line logs
- "trace" debug entries, usable by ``--loglevel=trace``, which outputs things like WAMP calls routed, for super debugging.

cc: @oberstet

### CODE REVIEW

File: crossbar/_logging.py [hunk 1]
```diff
@@ -150,6 +153,70 @@ def StandardErrorObserver(event):
...
+
+    recordSeparator = u"\x1e"
+    return FileLogObserver(
+        outFile,
+        lambda event: u"{0}{1}".format(_make_json(event), recordSeparator)
+    )
+
+
+def makeLegacyDailyLogFileObserver(path, logoutputlevel):
+    """
```
Comment (meejah) [inline crossbar/_logging.py]: everything else in crossbar uses under_score style names; maybe more consistent to do so with this _logging module as well?

Comment (hawkowl) [inline crossbar/_logging.py]: True, I'll fix that up.

File: crossbar/controller/processtypes.py [hunk 2]
```diff
@@ -104,34 +107,23 @@ def log(self, childFD, data):
         """
         assert(childFD in self._log_fds)
 
-        if _PY3 and type(data) != str:
+        if type(data) != unicode:
             data = data.decode('utf8')
```
Comment (meejah) [inline crossbar/controller/processtypes.py]: I think `if type(data) != six.text_type` or `if not isinstance(data, six.text_type)` for consistency with rest of crossbar

**Comment by meejah** [comment]
To ditto my other comment about method names, I would change `camelCase` variable names to `under_score` style to match the rest of crossbar (especially `logPublisher` -> `log_publisher` since it's part of the API)

File: crossbar/controller/processtypes.py (hunk 2 above)
Comment (hawkowl) [inline crossbar/controller/processtypes.py]: Changed to use six instead

**Comment by hawkowl** [comment]
@meejah I think I've got all of the non-Twisted public API that `_logging` exposes in underscore style, in the latest diff.

**Comment by meejah** [comment]
cool, looks good. I didn't actually try running it, however ;)

**Comment by hawkowl** [comment]
I'll make that a variable that's imported, and provide a link to why it is that specifically (there's a JSON log stread RFC for it).

**Comment by hawkowl** [comment]
(I wish you could reply to comments in github :( )

**Comment by meejah** [comment]
yeah, threaded comments would be nice...

**Pull request merged**